src/test/javascript/spec/component/overview/course-lectures/course-lecture-details.component.spec.ts

```diff
@@ -0,0 +1,254 @@
+import { IncludedInScoreBadgeComponent } from 'app/exercises/shared/exercise-headers/included-in-score-badge.component';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+@Component({ selector: 'jhi-course-exercise-row', template: '' })
+class CourseExerciseRowStubComponent {
+    @Input() exercise: Exercise;
+    @Input() course: Course;
+    @Input() extendedLink = false;
+    @Input() hasGuidedTour: boolean;
+    @Input() isPresentationMode = false;
+}
+
+class MockActivatedRoute {
+    parent: any;
+    params: any;
+
+    constructor(options: { parent?: any; params?: any }) {
+        this.parent = options.parent;
+        this.params = options.params;
+    }
+}
+
+describe('CourseLectureDetails', () => {
+    let fixture: ComponentFixture<CourseLectureDetailsComponent>;
+    let courseLecturesDetailsComponent: CourseLectureDetailsComponent;
+    let lecture: Lecture;
+    let lectureUnit1: AttachmentUnit;
+    let lectureUnit2: AttachmentUnit;
+    let lectureUnit3: TextUnit;
+    let mockRouter: any;
+    let debugElement: DebugElement;
+
+    beforeEach(() => {
+        mockRouter = sinon.createStubInstance(Router);
+
+        const releaseDate = moment('18-03-2020', 'DD-MM-YYYY');
+
+        const course = new Course();
+        course.id = 456;
+
+        lecture = new Lecture();
+        lecture.id = 1;
+        lecture.startDate = releaseDate;
+        lecture.description = 'Test desciption';
+        lecture.title = 'Test lecture';
+        lecture.course = course;
+
+        const attachment = new Attachment();
+        attachment.id = 1;
+        attachment.version = 1;
+        attachment.attachmentType = AttachmentType.FILE;
+        attachment.releaseDate = moment({ years: 2020, months: 3, date: 5 });
+        attachment.uploadDate = moment({ years: 2020, months: 3, date: 5 });
+        attachment.name = 'test';
+        attachment.link = '/path/to/file/test.pdf';
+
+        lectureUnit1 = new AttachmentUnit();
+        lectureUnit1.id = 1;
+        lectureUnit1.name = 'Unit 1';
+        lectureUnit1.releaseDate = releaseDate;
+        lectureUnit1.lecture = lecture;
+        lectureUnit1.attachment = attachment;
+        attachment.attachmentUnit = lectureUnit1;
+
+        const attachment2 = new Attachment();
+        attachment2.id = 1;
+        attachment2.version = 1;
+        attachment2.attachmentType = AttachmentType.FILE;
+        attachment2.releaseDate = moment({ years: 2020, months: 3, date: 5 });
+        attachment2.uploadDate = moment({ years: 2020, months: 3, date: 5 });
+        attachment2.name = 'test';
+        attachment2.link = '/path/to/file/test.pdf';
+
+        lectureUnit2 = new AttachmentUnit();
+        lectureUnit2.id = 2;
+        lectureUnit2.name = 'Unit 2';
+        lectureUnit2.releaseDate = releaseDate;
+        lectureUnit2.lecture = lecture;
+        lectureUnit2.attachment = attachment2;
+        attachment2.attachmentUnit = lectureUnit2;
```

Comment:
Can you move this to a function that returns attachment unit so you don't need to write it twice?